lib/middleware/operation.js

```diff
@@ -1,48 +1,69 @@
-const debug = require('debug')('hydra-box:operation')
+const { error, warn } = require('../log')('operation')
 const clownface = require('clownface')
 const ns = require('@tpluscode/rdf-ns-builders')
 const { code } = require('../namespaces')
 
+function findClassOperations (types, method) {
+  return types
+    .out(ns.hydra.supportedOperation)
+    .has(ns.hydra.method, method)
+    .toArray()
+}
+
+function findPropertyOperations (req, types, method) {
+  const properties = [...req.hydra.resource.dataset
+    .match(req.hydra.resource.term, null, req.hydra.term)]
+    .map(({ predicate }) => predicate)
+
+  return types
+    .out(ns.hydra.supportedProperty)
+    .has(ns.hydra.property, properties)
+    .out(ns.hydra.property)
+    .out(ns.hydra.supportedOperation)
+    .has(ns.hydra.method, method)
+    .toArray()
+}
+
 function factory (api) {
   return (req, res, next) => {
     if (!req.hydra.resource) {
       return next()
     }
 
-    const operations = []
-
     const method = req.method === 'HEAD' ? 'GET' : req.method
+    const types = clownface({ ...api, term: [...req.hydra.resource.types] })
 
-    clownface({ ...api, term: [...req.hydra.resource.types] })
-      .out(ns.hydra.supportedOperation)
-      .has(ns.hydra.method, method).forEach(operation => {
-        operations.push(operation)
-      })
+    let operations = []
+    if (req.hydra.term.equals(req.hydra.resource.term)) {
+      // only look for direct operation when the root resource is requested
+      operations = findClassOperations(types, method)
+    }
+    if (!operations.some(Boolean)) {
```

Comment:
Why `!operations.some(Boolean)` and not `operations.length === 0`? I think there would be only a difference if `operations` contains boolean false values, but that should not happen. The `length` check would make it more obvious what the actual content of the array is.